### PR TITLE
var #2573 issue

### PR DESCRIPTION
**Pull request by Mohammad-Talaat7**

### CODE REVIEW

**Comment by Oxbeefbabe** [comment]
Hi there @Mohammad-Talaat7, are you an intern applicant for the coding challenge?

**Comment by Mohammad-Talaat7** [comment]
> Hi there @Mohammad-Talaat7, are you an intern applicant for the coding challenge?

Hello sir @Oxbeefbabe yes I am and here a screenshot of my email after i submitted my upwork url


if there's anything else i need to provide let me know. 
Waiting for your reply.
Best regards.

**Comment by Oxbeefbabe** [comment]
Looks all good. Before I accept and merge, can you please make sure that the "lint / check-formatting (pull_request)" tests are passing here on Github? It should just be some basic reformatting. Thanks :)

**Comment by Oxbeefbabe** [comment]
I've started the workflow now. 

It is asking for a maintainer to start the workflow in your case because you have just recently made a Github account a few days ago, and because you have no history on the account, there are extra steps. This is to stop people from making many pull requests and using lots of resources on the workflow queue. Apologies for the inconvenience, I'll keep a close eye on the PR and accept any workflow start requests for you when they come up.

**Comment by Oxbeefbabe** [comment]
Hi, I've started the workflow again. If you fetch from upstream it should fix all of the lint errors not affected by your changes.

For the frontend task, you need to implement the chosen frontend function/tests for all of the backends (Tensorflow/Jax/Numpy/Torch)

**Comment by Mohammad-Talaat7** [comment]
Hello Eng. @Oxbeefbabe, i updated and solved alot of stuff i would be glad if you can check them and in issue #2573 they are asking to only add the numpy var front end so shouldnt i only add the front end and the test to numpy only?
Thanks in advance
Best Regards

**Comment by Oxbeefbabe** [comment]
Sorry for keeping you waiting! This looks good, you just have a small lint error in one of your files. Can you please fix that, and then fetch from upstream and fix and merge conflicts, and then I should be able to accept this.

**Comment by Mohammad-Talaat7** [comment]
> Contributor



> Sorry for keeping you waiting! One of the tests is failing for NumPy. You only need to make sure the tests you added are passing. You also have a small lint error in one of your files. Can you please fix that, and then fetch from upstream and fix and merge conflicts, and then I should be able to accept this?

Hello Eng. Ludwig, can you start the work flow so i can check the lint error?

**Comment by Mohammad-Talaat7** [comment]
> 

@Oxbeefbabe and so i can see the test error in which line exactly

**Comment by Mohammad-Talaat7** [comment]
@Oxbeefbabe that should do it i think

**Comment by Mohammad-Talaat7** [comment]
@Oxbeefbabe i did all the changes needed can you start the workflow for me.
Thanks in advance.

**Comment by Oxbeefbabe** [comment]
I've started the workflow. You'll need to merge from upstream again as there are merge conflicts again.

**Comment by Mohammad-Talaat7** [comment]
> I've started the workflow. You'll need to merge from upstream again as there are merge conflicts again.

@Oxbeefbabe Hello again i solved the conflicts can you start the workflow again.
Thanks.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py
Comment (Oxbeefbabe) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py:65]: In test_averages_and_variances.py, Line 65, you have passed ddof a second time. You should remove this as you already pass ddof in Line 57.

**Review by Oxbeefbabe — CHANGES_REQUESTED** [review]
In test_averages_and_variances.py, Line 65, you have passed ddof a second time. You should remove this as you already pass ddof in Line 57.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py
Comment (Mohammad-Talaat7) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py]: i fixed it it should be fine now

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py
Comment (Oxbeefbabe) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py:49]: The test function you should be using is `test_fronend_function`. `test_function` is only for backend functions

**Review by Oxbeefbabe — CHANGES_REQUESTED** [review]
Test function line 49 you are using the wrong test function. See my comment on it for more info

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py
Comment (Mohammad-Talaat7) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py]: @Oxbeefbabe i resolved the issue

**Comment by Mohammad-Talaat7** [comment]
@Oxbeefbabe i resolved the issue it should work now i guess

**Comment by Oxbeefbabe** [comment]
Your test is failing. You can find more information in test-frontend-NumPy / run-nightly-tests: Line 1851 and below. See if you can get this fixed please :)

**Comment by Mohammad-Talaat7** [comment]
> Your test is failing. You can find more information in test-frontend-NumPy / run-nightly-tests: Line 1851 and below. See if you can get this fixed please :)

Hello Eng. Ludwig.
I'm kinda stuck and I'm not sure what to do can you give me any hint
I checked the error lines and managed to reduce the Hypothesis failures to only 2 on my tests but I seem to be stuck I dont really know what to do exactly to remove them

**Comment by Oxbeefbabe** [comment]
In test-frontend-NumPy / run-nightly-tests: Line 1917, the error shows that it is trying to concatenate a list to the end of a string. The string, in this case, is the label for the data type you are passing in. `handle_where_and_array_bools` assumes that `input_dtypes` is a list, because there are functions that can take multiple lists of different data types, and the function we use handles this by having each index of the generated flags correspond to a specific array that is passed to a function.

You need to change what you pass in to `handle_where_and_array_bools` for `input_dtypes`.

**Comment by Emperor-WS** [comment]
> 

@Oxbeefbabe so it should be like this?

**Comment by Mohammad-Talaat7** [comment]
> In test-frontend-NumPy / run-nightly-tests: Line 1917, the error shows that it is trying to concatenate a list to the end of a string. The string, in this case, is the label for the data type you are passing in. `handle_where_and_array_bools` assumes that `input_dtypes` is a list, because there are functions that can take multiple lists of different data types, and the function we use handles this by having each index of the generated flags correspond to a specific array that is passed to a function.
> 
> You need to change what you pass in to `handle_where_and_array_bools` for `input_dtypes`.

@Oxbeefbabe so if i changed it like this it should work?
input_dtype, x = dtype_and_x
    where = np_frontend_helpers.handle_where_and_array_bools(
        where=where,
        input_dtype=[input_dtype],
        as_variable=as_variable,
        native_array=native_array,
    )

**Comment by Mohammad-Talaat7** [comment]
@Oxbeefbabe i don't know why it did this but i tried to recommit 2 codes but it merged on its on do i need to open another PR? or we can reopen this one?

**Pull request merged**
